yoshi/yoshi.py

```diff
@@ -0,0 +1,72 @@
+import proseco
+
+
+def run_one_yoshi(**kw):
+    """
+    Run proseco and sparkles for an observation request in a roll/temperature/man_angle
+    scenario.
+
+    :param detector: detector (ACIS-I|ACIS-S|HRC-I|HRC-S)
+    :param chipx: chipx from zero-offset aimpoint table entry for obsid
+    :param chipy: chipy from zero-offset aimpoint table entry for obsid
+    :param chip_id: chip_id from zero-offset aimpoint table entry for obsid
+    :param ra_targ: target RA (degrees)
+    :param dec_targ: target Dec (degrees)
+    :param roll_targ: target Roll (degrees)
+    :param offset_y: target offset_y (arcmin)
+    :param offset_z: target offset_z (arcmin)
+    :param sim_offset: SIM Z offset (steps)
+    :param focus_offset: SIM focus offset (steps)
+    :param dither_y: Y amplitude dither (arcsec)
+    :param dither_z: Z amplitude dither (arcsec)
+    :param obs_date: observation date (for proper motion and ACA offset projection)
+    :param t_ccd: ACA CCD temperature (degrees C)
+    :param man_angle: maneuver angle (degrees)
+    :returns: dictionary of (ra_pnt, dec_pnt, roll_pnt,
+                             N_critical, N_warning, N_caution, N_info,
+                             P2, guide_count)
+
+    """
+
+    # Calculate dynamic offsets using the supplied temperature.
+    aca_offset_y, aca_offset_z = get_aca_offsets(kw['detector'],
+                                                 kw['chip_id'], kw['chipx'], kw['chipy'],
+                                                 kw['obs_date'], kw['t_ccd'])
+
+    # Get the ACA quaternion using target offsets and dynamic offsets.
+    # Note that calc_aca_from_targ expects target offsets in degrees and obs is now in arcmin
+    q_pnt = calc_aca_from_targ((kw['ra_targ'], kw['dec_targ'], kw['roll_targ']),
+                               (kw['offset_y'] / 60.) + (aca_offset_y / 3600.),
+                               (kw['offset_z'] / 60.) + (aca_offset_z / 3600.))
+
+    # Run proseco and sparkles
+    aca = proseco.get_aca_catalog(obsid=0,
+                                  att=q_pnt,
+                                  man_angle=kw['man_angle'],
+                                  date=kw['obs_date'],
+                                  t_ccd_acq=kw['t_ccd'],
+                                  t_ccd_guide=kw['t_ccd'],
+                                  dither_acq=(kw['dither_y'], kw['dither_z']),
+                                  dither_guide=(kw['dither_y'], kw['dither_z']),
+                                  detector=kw['detector'],
+                                  sim_offset=kw['sim_offset'],
+                                  focus_offset=kw['focus_offset'],
+                                  n_acq=8, n_guide=5, n_fid=3)
+    acar = aca.get_review_table()
+    acar.run_aca_review()
+
+    # Get values for report
+    report = {'ra_pnt': q_pnt.ra,
+              'dec_pnt': q_pnt.dec,
+              'roll_pnt': q_pnt.roll,
+              'N_critical': len(acar.messages == 'critical'),
```

Comment:
Should this also return a good/bad or tri-state or some such (so it can be easily controlled on this side)?

Reply:
The google doc described having a "yes/no" for a catalog.

Reply:
SOT MP should manage the status, they will manage the code.  Changes in the FOT MP process that might impact the way yoshi data are interpreted will be coordinated between FOT MP, SOT MP, ACA.  But there is no really good reason to have the yes/no be hardwired into the HDF5 outputs (and indeed the google doc does not imply that).

Reply:
I thought they might manage this code too.  I suppose a better question would be, should there be another function in here that returns a yes/no status given these outputs.  

And yes, I was not thinking about hardwiring in a yes/no value into the HDF5 table, though I think in many cases the outputs will need to be remade anyway.  Also, I made no assumptions here about what input values would be useful to save back into that table; I think that SOT MP may want to save version information (hash of input params or just date of processing) to allow calling code to figure out what needs to be rerun.

Reply:
SOT MP will probably write such a function, but it doesn't need to be provided in this code now.  Nominally the answer is `n_critical == 0`, right?  So do we need to define that function for them?  And of course with experience the definition might evolve.

Reply:
👍 I think the only issue I have with using `n_critical == 0` for this kind of application is that it conflates guide and acq issues.  But I agree that the definition can and will continue to evolve and it doesn't need to live here.